### DISCUSSION TITLE
How to force FluentMigrator to run VerionInfo table migrations in Preview Mode

### DISCUSSION THREAD

**Discussion by Beatles1692** [fluentmigrator/fluentmigrator#1274] [General] [answered]
I have already posted my question to SO :
[https://stackoverflow.com/questions/62340816/how-to-force-fluentmigrator-to-run-verioninfo-table-migrations](url)

**Comment by jzabroski** [comment, marked as answer]
Hi @Beatles1692 ,

This is currently being tracked in https://github.com/fluentmigrator/fluentmigrator/issues/1226

There are a couple of other relevant issues related to preview mode and versioninfo table:

1. https://github.com/fluentmigrator/fluentmigrator/issues/1047
2. https://github.com/fluentmigrator/fluentmigrator/issues/978
3. https://github.com/fluentmigrator/fluentmigrator/issues/708 - this is an older issue and some of the asks here are out-of-date

**Reply by Beatles1692** [reply]
So If I understaood it correctly , there is no way to achieve this right now ?

**Reply by jzabroski** [reply]
It all depends on what your needs are.  

If you're not using transactions, #1047 doesn't matter to you.
If you just want to seed the creation of the VersionInfo table, you can just add the create table command to the front of your deployment process.  Doing that workaround, #1226 doesn't effect you.
If you want to run specific migrations, then #1033 would solve your problem (it should probably be higher priority).

Also, on the extreme, you could write your own command line tool and stitch everything together.  It's not impossible, but it also has limitations, such as Conditionals not working correctly.

**Reply by jzabroski** [reply]
@Beatles1692 I created a ticker #1276 to track some of what we discussed.  Thanks for asking such focused questions, as it helps me to in turn focus on how to solve a couple of related problems all at once.  I spent a good chunk of today trying to write up a design document.

**Reply by danpadmore** [reply]
@jzabroski I saw the [Unit of work ADR](https://github.com/fluentmigrator/fluentmigrator/blob/master/adr/proposed/UnitOfWork.md), which looks to solve or improve many of the things discussed here and elsewhere related to preview mode.

I'm in a situation on my current project, where I am required to generate scripts for manual deployment (preview + no-connection), similar to the user stories outlined in the ADR.

This surfaces some issues like:
1. Generated script for Oracle is not a valid script to execute in its entirety in Oracle Sql Developer, need to manually execute migration per migration (or should this work and am I doing it wrong?)
2. Can't use Schema.Exists because this executes instead of generating script (love the mentioned [Preconditions](https://github.com/fluentmigrator/fluentmigrator/issues/1281) idea as a solution)
3. VersionInfo.AppliedOn contains script generation date, want date of script execution (e.g. something like GETDATE() instead of a hardcoded date in script)
4. Want to wrap generated script with a BEGIN TRANSACTION (currently targeting SQL Server and Oracle)
(5. Drop/create database to improve local developer experience - ok, not related to script generation :) )


Is there something I can do to help progress the UnitOfWork proposal?
I suspect this is a non trivial change in the design, but if you already have specific tasks, maybe it is realistic to delegate some work.

**Comment by jzabroski** [comment]
It's a non-trivial change. I had a hairbrained idea for a hacky solution
that might work faster: mocking DbConnection and DbTransaction so that
calls to CreateCommand to execute a query write to the log.
